### DISCUSSION TITLE
Azure devops pipeline for UNO, Uno.check takes a considerable time, any way to cache the "build environment" ?

### DISCUSSION THREAD

**Discussion by digger73** [unoplatform/uno#18885] [General]
Hi

I'm very new at using hosted Azure devops for CI, usually it have been Teamcity but for a new project where we are using UNO platform we are moving in the cloud.
I have succeeded in get the pipeline building by using the example project provided for the UNO solution. However, the "install-dependencies.yml" which is responsible for setting up UNO in the build agent takes a considerable time every time a pipeline is run (9min and up).
I have tried to figure out, if it is possible to "cache" the setup the build environments dependencies (UNO), but haven't had any luck by now.

Does anyone have any knowledge if it is possible to store / cache the "uno check" step so it isn't necessary to perform every time ?

**Comment by jeromelaban** [comment]
Thanks for the question. At this time, there's no way to "cache" the setup unfortunately, because the longest part of the tool is the .NET workloads and that is not cacheable. 

That being said, you can use a combination of the [`--skip` parameter](https://github.com/unoplatform/uno.check/blob/main/doc/using-uno-check.md#running-unocheck-in-a-ci-environment) to remove some checks that take time, but you can also use [the `--target` parameter](https://github.com/unoplatform/uno.check/blob/main/doc/using-uno-check.md#command-line-arguments) to only install what you need.
